### PR TITLE
Fix documentation

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

**Comment by chfast** [comment]
Work but waiting for Docker Hub, and "docs" branch to be removed from whitelist.

File: include/evmc.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/// EVM-C -- C interface to Ethereum Virtual Machine
+/// EVMC -- Ethereum Client-VM Connector API
```
Comment (axic) [inline include/evmc.h]: I felt like merging #2 was a bit rushed :)

File: circle.yml [hunk 2]
```diff
@@ -14,8 +14,50 @@ jobs:
...
+      - checkout
+      - run:
+          name: "Generate documentation"
+          command: doxygen docs/Doxyfile
+      - run:
+          name: "Upload documentation"
+          command: |
+            git config user.email "docs-bot@ethereum.org"
+            git config user.name "Documentation Bot"
+            git add docs
```
Comment (axic) [inline circle.yml]: Actually can you move it to the root instead so that https://ethereum.github.io/evmc works.

Comment (chfast) [reply circle.yml]: I will try.

Comment (axic) [reply circle.yml]: Both work for me equally.

File: include/evmc.h [hunk 3]
```diff
@@ -273,8 +273,8 @@ struct evmc_result {
 /// @see evmc_get_optional_data(), evmc_get_const_optional_data().
 union evmc_result_optional_data
```
Comment (axic) [inline include/evmc.h]: Is this actually still used?!

Comment (chfast) [reply include/evmc.h]: Not sure. I will check it.

**Pull request merged**